lib/chef/provider/git.rb

```diff
@@ -154,6 +154,14 @@ def find_current_revision
         sha_hash?(result) ? result : nil
       end
 
+      def already_on_target_branch?
```

Comment:
This function could take in the target branch and just look like this:
```
def already_on_target_branch?(branch)
  git("rev-parse", "--abbrev-ref", "HEAD", cwd: cwd, returns: [0, 128]).stdout.strip == branch
end
```

Or if you don't want to have any inputs you only need to write the git command a single time which makes for less repetitive code:
```
current_branch = git("rev-parse", "--abbrev-ref", "HEAD", cwd: cwd, returns: [0, 128]).stdout.strip
if new_resource.checkout_branch
  current_branch == new_resource.checkout_branch
else
  current_branch == new_resource.revision
end
```

Reply:
I think it is easier to read if no argument to that branch. I made the requested change !